### PR TITLE
feat: Save a data snapshot as a CSV file

### PR DESCRIPTION
**Pull request by arkadyan**
Asana ticket: [Ability to save measurement snapshots across a range of parameter values](https://app.asana.com/0/1167196461144361/1204649882225658)

Capture prediction result stats for the current state of a delay alert algorithm across a range of parameters defined by the algorithm itself.

### CODE REVIEW

**Comment by arkadyan** [comment]
@ErinLMoore I stumbled my way into this approach after a couple of false starts. I think it’s fairly reasonable in the end, but definitely let me know if you can think of better ways to approach any of this.

File: lib/alerts_viewer_web/live/delay_alert_algorithm/median_component.ex [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
   Component for controlling the Median delay alert recommendation algorithm.
   """
 
   use AlertsViewerWeb, :live_component
 
+  @behaviour AlertsViewer.DelayAlertAlgorithm
+
   alias Routes.{Route, RouteStats}
 
+  @snapshot_median_min 50
```
Comment (ErinLMoore) [inline lib/alerts_viewer_web/live/delay_alert_algorithm/median_component.ex:12]: We can probably use these values on lines 45 and 46 too

**Comment by ErinLMoore** [comment]
> @ErinLMoore I stumbled my way into this approach after a couple of false starts. I think it’s fairly reasonable in the end, but definitely let me know if you can think of better ways to approach any of this.

I think I like the approach for now, I think I'll have to see what it's like with a second algorithm, whether it's easier to keep it like this or to give each algorithm their own csv functions (like maybe a csv behavior they implement or something). I think they're gonna be similar enough where this is fine, though.

File: lib/alerts_viewer_web/controllers/bus_controller.ex [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    mod = String.to_existing_atom(algorithm)
+    bus_routes = Routes.all_bus_routes()
+    stats_by_route = RouteStatsPubSub.all()
+    bus_alerts = Alerts.all()
+    routes_with_current_alerts = Enum.filter(bus_routes, &delay_alert?(&1, bus_alerts))
+
+    algoritm_data = mod.snapshot(bus_routes, stats_by_route)
+
+    data =
+      algoritm_data
```
Comment (ErinLMoore) [inline lib/alerts_viewer_web/controllers/bus_controller.ex:18]: small typo (algoritm -> algorithm)

File: lib/alerts_viewer_web/live/delay_alert_algorithm/median_component.ex (hunk 1 above)
Comment (arkadyan) [reply lib/alerts_viewer_web/live/delay_alert_algorithm/median_component.ex:12]: Yep, that's a good point. Wired up a couple private functions to pipe those through.

**Comment by arkadyan** [comment]
@ErinLMoore Yep, I agree 100%. Let's see what it feels like with one or two more examples and if it turns out to be awkward I'm by no means wedded to it.

**Review by ErinLMoore — APPROVED** [review]
Left a couple small comments but I like the overall approach. 🆒

**Pull request merged**